src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -348,6 +409,38 @@ private void EnsureSignature()
             }
         }
 
+        private void EnsureBackingFieldIsSynthesized()
```

Comment:
>EnsureBackingFieldIsSynthesized [](http://example.com/codeflow?start=21&length=31)

Consider also checking if ```_lazyBackingFieldSymbolForFieldKeyword != _lazyBackingFieldSymbolForFieldKeywordSentinel``` on entry and returning early. Otherwise we will be doing extra work every time ```FieldKeywordBackingField``` is accessed.